bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/BigtableSyncTableAccessor.java

```diff
@@ -0,0 +1,150 @@
+    syncTable.targetZkCluster = targetZkCluster;
+  }
+
+  public static String getTargetTableName(SyncTable syncTable) {
+    return syncTable.targetTableName;
+  }
+
+  public static void setTargetTableName(SyncTable syncTable, String targetTableName) {
+    syncTable.targetTableName = targetTableName;
+  }
+
+  public static String getConfDryRunKey() {
+    return SyncTable.DRY_RUN_CONF_KEY;
+  }
+
+  public static void setDryRun(SyncTable syncTable, boolean dryRun) {
+    syncTable.dryRun = dryRun;
+  }
+
+  public static void setDoDeletes(SyncTable syncTable, boolean doDeletes) {
+    syncTable.doDeletes = doDeletes;
+  }
+
+  public static void setDoPuts(SyncTable syncTable, boolean doPuts) {
+    syncTable.doPuts = doPuts;
+  }
+
+  public static void setIgnoreTimestamps(SyncTable syncTable, boolean ignoreTimestamps) {
+    syncTable.ignoreTimestamps = ignoreTimestamps;
+  }
+
+  public static Connection getSourceConnection(SyncMapper mapper) {
+    return mapper.sourceConnection;
+  }
+
+  public static Table getSourceTable(SyncMapper mapper) {
+    return mapper.sourceTable;
+  }
+
+  public static Connection getTargetConnection(SyncMapper mapper) throws IOException {
+    return mapper.targetConnection;
+  }
+
+  public static Table getTargetTable(SyncMapper mapper) {
+    return mapper.targetTable;
+  }
+
+  public static String getTargetZkClusterConf(Configuration conf) {
+    return conf.get(SyncTable.TARGET_ZK_CLUSTER_CONF_KEY);
+  }
+
+  public static String getSourceZkClusterConf(Configuration conf) {
+    return conf.get(SyncTable.SOURCE_ZK_CLUSTER_CONF_KEY);
+  }
+
+  public static void setTargetZkClusterConf(Configuration conf, String targetZkClusterConf) {
+    conf.set(SyncTable.TARGET_ZK_CLUSTER_CONF_KEY, targetZkClusterConf);
+  }
+
+  public static void setSourceZkClusterConf(Configuration conf, String sourceZkClusterConf) {
+    conf.set(SyncTable.SOURCE_ZK_CLUSTER_CONF_KEY, sourceZkClusterConf);
+  }
+
+  public static Connection setSourceConnection(SyncMapper mapper, Configuration conf)
+      throws IOException {
+    mapper.sourceConnection = ConnectionFactory.createConnection(conf);
+    return mapper.sourceConnection;
+  }
+
+  public static void setSourceTable(SyncMapper mapper, Connection connection, Configuration conf)
+      throws IOException {
+    mapper.sourceTable =
+        connection.getTable(TableName.valueOf(conf.get(SyncTable.SOURCE_TABLE_CONF_KEY)));
+  }
+
+  public static Connection setTargetConnection(SyncMapper mapper, Configuration conf)
+      throws IOException {
+    mapper.targetConnection = ConnectionFactory.createConnection(conf);
+    return mapper.targetConnection;
+  }
+
+  public static void setTargetTable(SyncMapper mapper, Connection connection, Configuration conf)
+      throws IOException {
+    mapper.targetTable =
+        connection.getTable(TableName.valueOf(conf.get(SyncTable.TARGET_TABLE_CONF_KEY)));
+  }
```

Comment:
I think you should simplify this to just give you the conf keys and use those keys to access the configuration elsewhere.

Reply:
great suggestion, this cleaned things up a lot. updated, ptal